include/revng/Support/IRHelperRegistry.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/IR/Module.h"
+
+#include "revng/Support/ManagedStaticRegistry.h"
+
+struct RegisteredIRHelper {
+public:
+  std::string Name;
+  std::string Description;
+
+public:
+  const std::string &key() const { return Name; }
+};
+using RegisterIRHelper = RegisterManagedStaticImpl<RegisteredIRHelper>;
+
+inline const llvm::Function *getRegisteredIRHelper(llvm::StringRef Name,
+                                                   const llvm::Module &Module) {
+  const RegisteredIRHelper *Registered = RegisterIRHelper::get(Name);
+  if (not Registered) {
+    std::string Error = "'" + Name.str()
+                        + "' is not a known helper.\n"
+                          "Did you forget to register it?";
+    revng_abort(Error.c_str());
+  }
+
+  return Module.getFunction(Registered->Name);
+}
+
+inline llvm::Function *getRegisteredIRHelper(llvm::StringRef Name,
+                                             llvm::Module &Module) {
+  const RegisteredIRHelper *Registered = RegisterIRHelper::get(Name);
+  if (not Registered) {
+    std::string Error = "'" + Name.str()
+                        + "' is not a known helper.\n"
+                          "Did you forget to register it?";
+    revng_abort(Error.c_str());
+  }
+
+  return Module.getFunction(Registered->Name);
+}
+
+template<typename... Types>
+llvm::FunctionCallee getOrInsertRegisteredIRHelper(llvm::StringRef Name,
+                                                   llvm::Module &Module,
+                                                   Types &&...Values) {
+  const RegisteredIRHelper *Registered = RegisterIRHelper::get(Name);
+  if (not Registered) {
+    std::string Error = "'" + Name.str()
+                        + "' is not a known helper.\n"
+                          "Did you forget to register it?";
+    revng_abort(Error.c_str());
+  }
+
+  return Module.getOrInsertFunction(Name, std::forward<Types>(Values)...);
+}
+
+template<typename... Types>
```

Comment:
Please stop using parameter packs, they make the code harder to understand.

Reply:
(closing this, we discussed to further in a different thread)